accounts/kv_sql_migration_test.go

```diff
@@ -0,0 +1,339 @@
+				// Create an account with balance
+				acct1, err := kvStore.NewAccount(
+					ctxc, 0, time.Time{}, "foo",
+				)
+				require.NoError(t, err)
+				require.False(t, acct1.HasExpired())
+
+				hash1 := lntypes.Hash{1, 1, 1, 1}
+				known, err := kvStore.UpsertAccountPayment(
+					ctxc, acct1.ID, hash1, 100,
+					lnrpc.Payment_UNKNOWN,
+				)
+				require.NoError(t, err)
+				require.False(t, known)
+
+				hash2 := lntypes.Hash{2, 2, 2, 2}
+				known, err = kvStore.UpsertAccountPayment(
+					ctxc, acct1.ID, hash2, 200,
+					lnrpc.Payment_IN_FLIGHT,
+				)
+				require.NoError(t, err)
+				require.False(t, known)
+
+				hash3 := lntypes.Hash{3, 3, 3, 3}
+				known, err = kvStore.UpsertAccountPayment(
+					ctxc, acct1.ID, hash3, 200,
+					lnrpc.Payment_SUCCEEDED,
+				)
+				require.NoError(t, err)
+				require.False(t, known)
+
+				hash4 := lntypes.Hash{4, 4, 4, 4}
+				known, err = kvStore.UpsertAccountPayment(
+					ctxc, acct1.ID, hash4, 200,
+					lnrpc.Payment_FAILED,
+				)
+				require.NoError(t, err)
+				require.False(t, known)
+			},
+		},
+		{
+			"multiple accounts",
+			func(t *testing.T, kvStore *BoltStore) {
+				// Create two accounts with balance and that
+				// expires.
+				acct1, err := kvStore.NewAccount(
+					ctxc, 100000, time.Now().Add(time.Hour),
+					"foo",
+				)
+				require.NoError(t, err)
+				require.False(t, acct1.HasExpired())
+
+				acct2, err := kvStore.NewAccount(
+					ctxc, 200000, time.Now().Add(time.Hour),
+					"bar",
+				)
+				require.NoError(t, err)
+				require.False(t, acct2.HasExpired())
+
+				// Create invoices for both accounts.
+				hash1 := lntypes.Hash{1, 1, 1, 1}
+				err = kvStore.AddAccountInvoice(
+					ctxc, acct1.ID, hash1,
+				)
+				require.NoError(t, err)
+
+				hash2 := lntypes.Hash{2, 2, 2, 2}
+				err = kvStore.AddAccountInvoice(
+					ctxc, acct2.ID, hash2,
+				)
+				require.NoError(t, err)
+
+				// Create payments for both accounts.
+				hash3 := lntypes.Hash{3, 3, 3, 3}
+				known, err := kvStore.UpsertAccountPayment(
+					ctxc, acct1.ID, hash3, 100,
+					lnrpc.Payment_SUCCEEDED,
+				)
+				require.NoError(t, err)
+				require.False(t, known)
+
+				hash4 := lntypes.Hash{4, 4, 4, 4}
+				known, err = kvStore.UpsertAccountPayment(
+					ctxc, acct2.ID, hash4, 200,
+					lnrpc.Payment_IN_FLIGHT,
+				)
+				require.NoError(t, err)
+				require.False(t, known)
+			},
+		},
+	}
+
+	for _, test := range tests {
+		tc := test
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			var kvStore *BoltStore
+			testClock := clock.NewTestClock(time.Now())
+
+			kvStore, err := NewBoltStore(
+				t.TempDir(), DBFilename, testClock,
+			)
+			require.NoError(t, err)
+
+			tc.populateDB(t, kvStore)
+
+			t.Cleanup(func() {
+				require.NoError(t, kvStore.db.Close())
+			})
+
+			t.Run("Postgres", func(t *testing.T) {
+				migrationTest(t, kvStore, testClock, false)
+			})
+
+			t.Run("SQLite", func(t *testing.T) {
+				migrationTest(t, kvStore, testClock, true)
```

Comment:
in Lit, we've gone with the build flag approach. So can just use the existing NewTestDB methods no? and then if bbolt backend, just skip the test ?

Reply:
Updated as discussed offline :)